src/components/AllocationOverview.js

```diff
@@ -213,18 +213,20 @@ const AllocationOverview = (props) => {
     if (errorAllocation || errorClientPayments) return `Error`
 
     const { getAllocationById: allocation } = dataAllocation
-    const payments = [null]
-    if (clientPayments) {
-        payments.unshift(...clientPayments.payments)
+    const payments = dataClientPayments
+        ? [...dataClientPayments.getClientById.payments].sort((a, b) => b.date_incurred - a.date_incurred)
+        : [null]
+    if (payments.length) {
+        payments.push(null)
     }
 
     if (!contributorAllocation) {
         setContributorAllocation(allocation)
     }
     
     const editButtonDisabled = (
-        updatedAllocationRate.total_amount == allocationInfo.amount &&
-        updatedAllocationRate.hourly_rate == allocationInfo.rate.hourly_rate
+        updatedAllocationRate.total_amount <= 0 ||
+        updatedAllocationRate.hourly_rate <= 0
```

Comment:
1. We should allow hourly rates of zero, so we should only check for negative numbers
2. I would go ahead and try a different implementation to make this logic more robust. Instead of adding logic to check the total_amount and  hourly_rate in here, we shouldn't let the user write negative values or letters at the text fields in the first place, so go deeper into the components of the `RateProratedMonthlyForm` and `RateMaxBudgetForm` and try to implement a logic that will only change the value of the field if it's
* a non-negative number
* a number value
so something like this could work
```
const changeMonthlyHoursInput = (value) => {
        if (Number(value) >= 0 || null) {
            setMonthlyhoursInput(value)
        }
    }
```
but for every text field that you think it's needed.
Then we shouldn't need to place this logic in here

Reply:
I'm going to fix those Issues with Issue #480. For now, I'm going to revert the changes of how It was before.